### PR TITLE
Added Function96v2 Keyboard

### PR DESCRIPTION
**Pull request by MatthieuLu**
## Description

Added the firmware files for the function96v2 board, and all files compile properly.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/function96v2/keymaps/ansi_splitspace/keymap.c [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______
+      ),
+    [2] = LAYOUT_ansi_splitspace(
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______
+      )
```
Comment (fauxpark) [inline keyboards/function96v2/keymaps/ansi_splitspace/keymap.c:43]:
```suggestion
```

File: keyboards/function96v2/keymaps/ansi_splitspace/keymap.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+// Hello World
```
Comment (fauxpark) [inline keyboards/function96v2/keymaps/ansi_splitspace/keymap.c:46]:
```suggestion
```

File: keyboards/function96v2/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______
+      ),
+};
+
+// Hello World
```
Comment (fauxpark) [inline keyboards/function96v2/keymaps/default/keymap.c:38]:
```suggestion
};
```

File: keyboards/function96v2/keymaps/iso/keymap.c [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______
+      ),
+};
+
+// Hello World
```
Comment (fauxpark) [inline keyboards/function96v2/keymaps/iso/keymap.c:38]:
```suggestion
};
```

File: keyboards/function96v2/keymaps/iso_splitspace/keymap.c [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______
+      ),
+};
+
+// Hello World
```
Comment (fauxpark) [inline keyboards/function96v2/keymaps/iso_splitspace/keymap.c:46]:
```suggestion
};
```

File: keyboards/function96v2/readme.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# Function96v2
+
+This Function96v2 is a passion project of JTM. It's meant to give the user the maximum usability of the space allocated by the keyboard while also giving the option of using the numpad.
+
+* Keyboard Maintainer: The QMK Community
```
Comment (fauxpark) [inline keyboards/function96v2/readme.md:5]:
```suggestion
* Keyboard Maintainer: The QMK Community
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]
Whoops, actually the lint is complaining about these:

```
 ☒ function96v2: LAYOUT_default: Number of elements in info.json does not match! info.json:100 != LAYOUT_default:98
☒ function96v2: LAYOUT_ansi_splitspace: Number of elements in info.json does not match! info.json:102 != LAYOUT_ansi_splitspace:100
```

File: keyboards/function96v2/readme.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+* Hardware Supported: Function96v2
+* Hardware Availability: Limited Groupbuy that ended on 11/29/2020, https://www.reddit.com/r/mechmarket/comments/jxvokl/gb_function96v2_a_premium_96_layout_nov_20th_nov/
+
+Make example for this keyboard (after setting up your build environment):
+
+    make function96v2:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (drashna) [inline keyboards/function96v2/readme.md:13]: Could you add instructions on how to enter the bootloader?

**Comment by noroadsleft** [comment]
As this is a new revision of an existing keyboard design, you can and should set this up to share code across both v1 and v2 boards.

https://github.com/MatthieuLu/qmk_firmware/compare/d3845f15d899e4670ac245b5b3ba788ca6ee8cde...noroadsleft:c31136602bca520d7728a69d497e03e41c8b5377

**Pull request closed without merging**